### PR TITLE
[feat]: [CDS-68118]: Onboarded updateGitMetadata apis for pipeline and inputSets for newApiFlow

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes
Changes: Have onboarded update APIs for git-metadata.
Testing: Have tested pipeline and inputsets update flow with valid, empty and null arguments.(LOCAL)

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47575)

### CODE REVIEW

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -549,6 +549,62 @@ paths:
...
+      requestBody:
+        $ref: '#/components/requestBodies/GitMetadataUpdateRequestBody'
+      security:
+        - x-api-key: []
+      parameters:
+        - $ref: '#/components/parameters/AccountHeader'
+      tags:
+        - Pipelines
+      description: Update git-metadata in remote pipeline
+  '/v1/orgs/{org}/projects/{project}/pipelines/{pipeline}/input-sets/{input-set}/update/git-metadata':
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:574]: same as above

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -549,6 +549,62 @@ paths:
...
+    parameters:
+      - name: org
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:576]: please make these refs as you had done in the above api
- $ref: '#/components/parameters/OrgParam'

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -549,6 +549,62 @@ paths:
...
+      - $ref: '#/components/parameters/PipelineParam'
+    put:
+      summary: Update GitMetadata for Remote InputSet
+      operationId: update-git-metadata
+      responses:
+        '200':
+          $ref: '#/components/responses/GitMetadataUpdateResponseBody'
+      requestBody:
+        $ref: '#/components/requestBodies/GitMetadataUpdateRequestBody'
+      security:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:603]: missing account header

**Review by mankrit-singh — COMMENTED** [review]
reviewed spec changes with few comments

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47575)* status: 21 file(s) to review. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 552 at r2](https://reviewable.io/reviews/harness/harness-core/47575#gh-1193402353:-NVTLMfV51Ew_1dzJ1gN:b-g2sp12) ([raw file](https://github.com/harness/harness-core/blob/d74994f196572a4f3e8b468045f0fc642329663e/pipeline-service/contracts/openapi/v1/openapi.yaml#L552)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

why do we need 'update' here? PUT call and descriptions are enough to define the function ig
</blockquote></details>

just to be in sync with the old Apis
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 563 at r2](https://reviewable.io/reviews/harness/harness-core/47575#gh-1193402613:-NVTmknY6DGUf9YBsESz:b-hfpv0l) ([raw file](https://github.com/harness/harness-core/blob/d74994f196572a4f3e8b468045f0fc642329663e/pipeline-service/contracts/openapi/v1/openapi.yaml#L563)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

is the blank status code response needed?
</blockquote></details>

removed
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 574 at r2](https://reviewable.io/reviews/harness/harness-core/47575#gh-1193403196:-NVTMF_GAGHJCUpfAAJj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d74994f196572a4f3e8b468045f0fc642329663e/pipeline-service/contracts/openapi/v1/openapi.yaml#L574)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 576 at r2](https://reviewable.io/reviews/harness/harness-core/47575#gh-1193403673:-NVTMF2o88Nha3UIOAb6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d74994f196572a4f3e8b468045f0fc642329663e/pipeline-service/contracts/openapi/v1/openapi.yaml#L576)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

please make these refs as you had done in the above api
- $ref: '#/components/parameters/OrgParam'
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 603 at r2](https://reviewable.io/reviews/harness/harness-core/47575#gh-1193404337:-NVTMDtm0S--dFFIMiFx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d74994f196572a4f3e8b468045f0fc642329663e/pipeline-service/contracts/openapi/v1/openapi.yaml#L603)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

missing account header
</blockquote></details>

Done.

**Comment by senjucanon2** [comment]
trigger uts

**Comment by senjucanon2** [comment]
trigger codeformat

**Comment by senjucanon2** [comment]
trigger uts

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 1 of 20 files at r1, 13 of 17 files at r2, 1 of 3 files at r3, 4 of 5 files at r4, 1 of 1 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47575)* status: 21 file(s) to review. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger ut0

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java` line 231 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPMGl2y1uSFLKX6W2:-NVYPMGl2y1uSFLKX6W3:b-mey5md) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java#L231)):*
> ```Java
>   @Override
>   @NGAccessControlCheck(resourceType = "PIPELINE", permission = PipelineRbacPermissions.PIPELINE_CREATE_AND_EDIT)
>   public Response updateInputSetGitMetadata(@ResourceIdentifier @NotNull String pipeline, @OrgIdentifier String org,
> ```

parameters are not in order with respect to same API in pipeline and input-sets. Please make them consistent and in a proper order

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 219 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPhGn9hTT6xUZhpv-:-NVYPhGn9hTT6xUZhpv0:bu78mio) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L219)):*
> ```Java
>             .build());
> 
>     GitMetadataUpdateResponseBody gitMetadataUpdateResponseBody = new GitMetadataUpdateResponseBody();
> ```

there's no builder here?

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 212 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPjUtBdO2_T7vNO91:-NVYPjUtBdO2_T7vNO92:bopi57t) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L212)):*
> ```Java
>       @ResourceIdentifier String pipeline, @Valid GitMetadataUpdateRequestBody body,
>       @AccountIdentifier String harnessAccount) {
>     String pipelineIdentifierAfterUpdate = pmsPipelineService.updateGitMetadata(harnessAccount, org, project, pipeline,
> ```

variable name doesn't sound correct.

**Review by senjucanon2 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47575#-:-NVYVGWq0elRfA4XRSKI:bnfp4nl)* status: 21 file(s) to review. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java` line 231 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPMGl2y1uSFLKX6W2:-NVYWte22B9magEAd5-v:b-j3h35b) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java#L231)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

parameters are not in order with respect to same API in pipeline and input-sets. Please make them consistent and in a proper order
</blockquote></details>

I cannot determine this order it is auto-generated by StoplightStudio, in InputSets there is one extra parameter "pipeline" as Query params, and in stoplight query params are placed before path params.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 212 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPjUtBdO2_T7vNO91:-NVYXFTY0iKGAxdnxsDZ:b499q7j) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L212)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

variable name doesn't sound correct.
</blockquote></details>

Changed
___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 219 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPhGn9hTT6xUZhpv-:-NVYXN0_AfXg6gUyl03c:bc3gv3v) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L219)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

there's no builder here?
</blockquote></details>

Models created from Stoplight studio for new APIs do not support 3rd party annotations
___
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/GitMetadataUpdateInfoDTO.java` line 29 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYP9rG1jciN_c0qDu6:-NVYVJd23GgMOHtTcWZH:b-4pzkly) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/900-git-sync-sdk/src/main/java/io/harness/gitsync/GitMetadataUpdateInfoDTO.java#L29)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

This class shouldn't be in 900-git-sync-sdk, please move it from here.
</blockquote></details>

This is a trash class made in first attempt to write APIs removed.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java` line 231 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPMGl2y1uSFLKX6W2:-NVYa_Uc0FaqDWOpNekb:bvdtrgb) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java#L231)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

I cannot determine this order it is auto-generated by StoplightStudio, in InputSets there is one extra parameter "pipeline" as Query params, and in stoplight query params are placed before path params.
</blockquote></details>

@mankrit-singh params will be random? If I generate again, the order will change again?

**Comment by mankrit-singh** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java` line 231 at r7](https://reviewable.io/reviews/harness/harness-core/47575#-NVYPMGl2y1uSFLKX6W2:-NVhJ4kaAj87WL5gACpk:bpbaqbt) ([raw file](https://github.com/harness/harness-core/blob/72f0aad839115f557b7f0ac997e56af85959d488/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/api/InputSetsApiImpl.java#L231)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

@mankrit-singh params will be random? If I generate again, the order will change again?
</blockquote></details>

Checking this
I'm sure that on rebuilding the interfaces the order isnt changed, otherwise the APIs would break frequently.

**Comment by senjucanon2** [comment]
trigger uts

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 13 of 17 files at r2, 4 of 5 files at r4, 1 of 1 files at r6, 1 of 1 files at r7, 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47575)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger review

**Review by mankrit-singh — APPROVED** [review]

**Comment by mankrit-singh** [comment]
trigger review

**Pull request merged**